punx/tests/test_validate.py

```diff
@@ -654,4 +654,54 @@ def TODO_test_default_plot_v1_fail(hfile):
     # TODO:
 
 
+@pytest.mark.parametrize(
+    "infile, report, observations",
+    [
+        ["writer_1_3.hdf5", "TODO", 7],
+        ["writer_1_3.hdf5", "NOTE", 1],
+        ["writer_1_3.hdf5", "NOTE,TODO", 7 + 1],
+        ["writer_2_1.hdf5", "note", 0],
+        ["writer_2_1.hdf5", "TODO", 11],
+        ["1998spheres.h5", "ERROR", 2],
+        ["02_03_setup.h5", "NOTE,OPTIONAL,ERROR", 98 + 70 + 0],
+        ["prj_test.nexus.hdf5", "", 121],
+    ],
+)
+def test_report_option(infile, report, observations, capsys):
```

Comment:
What value does `capsys` get if pytest only passes three arguments?

Reply:
It's a pytest [fixture](https://docs.pytest.org/en/6.2.x/capture.html#accessing-captured-output-from-a-test-function).  It supplies the pytest feature that captures output to stdout & stderr.